File: CS/GrowthWare.WebApplication/Controllers/FileManagerController.cs
using GrowthWare.Framework.Common;
using GrowthWare.Framework.Model.Profiles;
using GrowthWare.WebSupport.Utilities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Net.Http;
using System.Web;
using System.Web.Http;

namespace GrowthWare.WebApplication.Controllers
{
    public class FileManagerController : ApiController
    {
        [HttpPost()]
        public IHttpActionResult CreateDirectory(string currentDirectoryString, int functionSeqId, string newDirectory) 
        {
            string mRetVal = "Unable to create directory";
            if (String.IsNullOrEmpty(currentDirectoryString) || String.IsNullOrEmpty(newDirectory))
            {
                mRetVal = "All parameters must be passed!";
                ArgumentException ex = new ArgumentException(mRetVal);
                Logger mLog = Logger.Instance();
                mLog.Error(mRetVal);
                throw (ex);
            }
            HttpServerUtility mServer = HttpContext.Current.Server;
            MDirectoryProfile mDirectoryProfile = DirectoryUtility.GetProfile(functionSeqId);
            String mCurrentDirectory = mDirectoryProfile.Directory;
            if (currentDirectoryString.Length > 0)
            {
                mCurrentDirectory += @"\" + currentDirectoryString;
            }
            mRetVal = FileUtility.CreateDirectory(mServer.UrlDecode(mCurrentDirectory), mServer.UrlDecode(newDirectory), mDirectoryProfile);
            return this.Ok(mRetVal);
        }

        [HttpPost()]
        public IHttpActionResult DeleteFiles(List<UIFileInfo> filesToDelete) 
        {
            HttpServerUtility mServer = HttpContext.Current.Server;
            string mRetVal = "Done";
            MDirectoryProfile mDirectoryProfile = DirectoryUtility.GetProfile(filesToDelete[0].FunctionSeqId);
            bool mExitLoop = false;
            foreach (UIFileInfo item in filesToDelete)
            {
                string mCurrentDirectory = mServer.UrlDecode(item.CurrentDirectory);
                if (mCurrentDirectory.Length == 0)
                {
                    mCurrentDirectory = mDirectoryProfile.Directory;
                }
                else
                {
                    mCurrentDirectory = mDirectoryProfile.Directory + mCurrentDirectory;
                }
                switch (item.FileType)
                {
                    case "File":
                        string mFileName = mCurrentDirectory += "/" + mServer.UrlDecode(item.FileName);
                        mRetVal = FileUtility.DeleteFile(mFileName, mDirectoryProfile);
                        if (mRetVal.IndexOf("Successfully") == -1) mExitLoop = true;
                        break;
                    case "Folder":
                        mCurrentDirectory += "/" + mServer.UrlDecode(item.FileName);
                        mRetVal = FileUtility.DeleteDirectory(mCurrentDirectory, mDirectoryProfile);
                        if (mRetVal.IndexOf("Successfully") == -1) mExitLoop = true;
                        break;
                    default:
                        break;
                }
                if (mExitLoop) break;
            }
            return this.Ok(mRetVal);
        }

        [HttpPost()]
        public IHttpActionResult GetDirectoryLinks(RequestDirectoryLinksInfo requestDirectoryInfo) 
        {
            string mRetVal = FileUtility.GetDirectoryLinks(requestDirectoryInfo.CurrentDirectoryString, requestDirectoryInfo.FunctionSeqId);
            return Ok(mRetVal);
        }
    }

    public class RequestDirectoryLinksInfo
    {
        public string CurrentDirectoryString { get; set; }
        public int FunctionSeqId { get; set; }
    }

    public class UIFileInfo
    {
        public string CurrentDirectory;
        public string FileName;
        public string FileType;
        public int FunctionSeqId;
    }
}

File: VB/GrowthWare.WebApplication/Controllers/FileManagerController.vb
Imports System.Collections.ObjectModel
Imports System.Net
Imports System.Web.Http
Imports GrowthWare.WebSupport.Utilities
Imports GrowthWare.Framework.Model.Profiles
Imports System.Web.Script.Serialization
Imports Newtonsoft.Json

Namespace Controllers
    Public Class FileManagerController
        Inherits ApiController

        <HttpPost>
        Public Function DeleteFiles(ByVal filesToDelete As List(Of UIFileInfo)) As IHttpActionResult
            Dim mServer As HttpServerUtility = HttpContext.Current.Server
            Dim mRetVal As String = "Done"
            Dim mDirectoryProfile As MDirectoryProfile = DirectoryUtility.GetProfile(filesToDelete(0).FunctionSeqId)
            For Each item As UIFileInfo In filesToDelete
                Dim mCurrentDirectory As String = mServer.UrlDecode(item.CurrentDirectory)
                If mCurrentDirectory.Length = 0 Then
                    mCurrentDirectory = mDirectoryProfile.Directory
                Else
                    mCurrentDirectory = mDirectoryProfile.Directory + mCurrentDirectory
                End If

                Select Case item.FileType
                    Case "File"
                        Dim mFileName As String = mCurrentDirectory + "\" + mServer.UrlDecode(item.FileName)
                        mRetVal = FileUtility.DeleteFile(mFileName, mDirectoryProfile)
                        If mRetVal.IndexOf("Successfully") = -1 Then Exit For
                    Case "Folder"
                        mCurrentDirectory = mCurrentDirectory + "/" + mServer.UrlDecode(item.FileName)
                        mRetVal = FileUtility.DeleteDirectory(mCurrentDirectory, mDirectoryProfile)
                        If mRetVal.IndexOf("Successfully") = -1 Then Exit For
                    Case Else

                End Select
            Next
            Return Me.Ok(mRetVal)
        End Function

        <HttpPost>
        Public Function GetDirectoryLinks(ByVal requestDirectoryInfo As RequestDirectoryLinksInfo) As IHttpActionResult
            Dim mRetVal As String = String.Empty
            mRetVal = FileUtility.GetDirectoryLinks(requestDirectoryInfo.CurrentDirectoryString, requestDirectoryInfo.FunctionSeqId)
            Return Me.Ok(mRetVal)
        End Function
    End Class

    Public Class RequestDirectoryLinksInfo
        Public Property CurrentDirectoryString() As String
        Public Property FunctionSeqId() As Integer
    End Class


    Public Class UIFileInfo
        Public CurrentDirectory As String
        Public FileName As String
        Public FileType As String
        Public FunctionSeqId As Integer
    End Class
End Namespace